giskard/client/giskard_client.py

```diff
@@ -261,6 +257,39 @@ def create_kernel(
             },
         )
 
+    def sync_kernel(
+        self,
+        kernel_name: str,
+        exact_deps: bool = False,
+        excludes: List[str] = EXCLUDED_PYLIBS,
+        only_giskard: bool = False,
+    ):
+        kernels = self._session.get("kernels").json()
```

Comment:
I think that it would be better to check the kernel by name instead of having to fetch the whole list (it's generally a bad practice to fetch a whole table when the data size isn't fixed):

```suggestion
        # TODO: handle 404 not found if kernel does not exists 
        kernel = self._session.get(f"kernels/{kernel_name}").json()
```